DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -17,21 +19,52 @@ public MyAccountController(ICustomPromptsService customPromptsService, IUserServ
         }
 
         public IActionResult Index()
+        {
+            var userAdminId = User.GetAdminId();
+            var userDelegateId = User.GetNullableCandidateId();
+            var (adminUser, delegateUser) = userService.GetUsersById(userAdminId, userDelegateId);
+
+            var customPrompts = customPromptsService.GetCustomPromptsForCentreByCentreId(delegateUser?.CentreId);
+
+            var model = new MyAccountViewModel(adminUser, delegateUser, customPrompts);
+
+            return View(model);
+        }
+
+        [HttpGet]
+        public IActionResult EditDetails()
         {
             if (!User.Identity.IsAuthenticated)
             {
                 return RedirectToAction("Index", "Login");
             }
 
-            var userAdminId = User.GetCustomClaim(CustomClaimTypes.UserAdminId);
-            var userDelegateId = User.GetCustomClaim(CustomClaimTypes.LearnCandidateId);
+            var userAdminId = User.GetAdminId();
+            var userDelegateId = User.GetNullableCandidateId();
             var (adminUser, delegateUser) = userService.GetUsersById(userAdminId, userDelegateId);
 
-            var customPrompts = customPromptsService.GetCustomPromptsForCentreByCentreId(delegateUser?.CentreId);
-
-            var model = new MyAccountViewModel(adminUser, delegateUser, customPrompts);
+            var model = new EditDetailsViewModel(adminUser, delegateUser);
 
             return View(model);
         }
+
+        [HttpPost]
+        public IActionResult EditDetails(EditDetailsViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            var signedInEmail = User.GetEmail();

```

Comment:
Edge case, but what if the user edits their email twice in one session (as we don't update the claims when updating the email address - or updating the email address from a different device)?

Reply:
I guess it's best to partially revert back to the previous flow then? Get Users by ID from claims IDs, get users by email from those users, update all these users?

Reply:
The concurrency issue with multiple devices is a bit more of a difficult issue to resolve, all I think we can do is just save whatever is latest since I don't think we have anything in place to handle that kind of issue

Reply:
The proposal of id -> email -> update sounds good and given that the user gets the latest email from the db, we shouldn't have issues with email being outdated